examples/DotNetCore/ChatApp/ChatApp/Program.cs

```diff
@@ -26,50 +27,80 @@
                });
 
         refresher = options.GetRefresher();
     })
     .Build();
 
 // Retrieve the OpenAI connection information from the configuration
-Uri openaiEndpoint = new (configuration["ChatApp:AzureOpenAI:Endpoint"]);
-string deploymentName = configuration["ChatApp:AzureOpenAI:DeploymentName"];
+var openAIConfiguration = configuration.GetSection("ChatApp:AzureOpenAI").Get<OpenAIConfiguration>();
 
-// Create a chat client
-AzureOpenAIClient azureClient = new(openaiEndpoint, credential);
-ChatClient chatClient = azureClient.GetChatClient(deploymentName);
+// Create a chat client using API key if available, otherwise use the DefaultAzureCredential
+AzureOpenAIClient azureClient;
+if (!string.IsNullOrEmpty(openAIConfiguration.ApiKey))
+{
+    azureClient = new AzureOpenAIClient(new Uri(openAIConfiguration.Endpoint), new Azure.AzureKeyCredential(openAIConfiguration.ApiKey));
+}
+else
+{
+    azureClient = new AzureOpenAIClient(new Uri(openAIConfiguration.Endpoint), credential);
+}
+ChatClient chatClient = azureClient.GetChatClient(openAIConfiguration.DeploymentName);
 
+// Initialize chat conversation
+var chatConversation = new List<ChatMessage>();
+Console.WriteLine("Chat started! What's on your mind?");
 while (true)
 {
     // Refresh the configuration from Azure App Configuration
     await refresher.TryRefreshAsync();
 
     // Configure chat completion with AI configuration
-    var modelConfiguration = configuration.GetSection("ChatApp:Model").Get<ModelConfiguration>();
+    var completionConfiguration = configuration.GetSection("ChatApp:Completion").Get<CompletionConfiguration>();
     var requestOptions = new ChatCompletionOptions()
     {
-        MaxOutputTokenCount = modelConfiguration.MaxTokens,
-        Temperature = modelConfiguration.Temperature,
-        TopP = modelConfiguration.TopP
+        MaxOutputTokenCount = completionConfiguration.MaxTokens,
+        Temperature = completionConfiguration.Temperature,
+        TopP = completionConfiguration.TopP
     };
 
-    foreach (var message in modelConfiguration.Messages)
+    // Get user input
+    Console.Write("You: ");
+    string? userInput = Console.ReadLine();
+
+    // Exit if user input is empty
+    if (string.IsNullOrEmpty(userInput))
     {
-        Console.WriteLine($"{message.Role}: {message.Content}");
+        Console.WriteLine("Exiting chat. Goodbye!");
+        break;
     }
 
-    // Get chat response from AI
-    var response = await chatClient.CompleteChatAsync(GetChatMessages(modelConfiguration), requestOptions);
-    System.Console.WriteLine($"AI response: {response.Value.Content[0].Text}");
+    // Add user message to chat conversation
+    chatConversation.Add(ChatMessage.CreateUserMessage(userInput));
+
+    // Get latest system message from AI configuration
+    var chatMessages = new List<ChatMessage>(GetChatMessages(completionConfiguration));
+    chatMessages.AddRange(chatConversation);

```

Comment:
This looks a bit off because it can add assistant and user messages after the existing user message from console. Perhaps GetChatMessages should be limitied to GetSystemMessages and then be prepended to the conversation

Reply:
Are you talking about the user message and assistant message a user may add in the AI configuration? They are added at the beginning of the message list `chatMessages` and the messgaes from console are added after. So after the first round of the conversation, they are just part of the conversation history. We need to update the ChatApp example in our document not to add a user message in the AI configuration.

Reply:
Discussed offline.